kernel/src/table_configuration.rs

```diff
@@ -317,36 +330,38 @@ impl TableConfiguration {
                 .unwrap_or(false)
     }
 
-    /// If in-commit timestamps is enabled, returns a tuple of the in-commit timestamp enablement
-    /// version and timestamp.
+    /// Returns information about in-commit timestamp enablement state.
     ///
-    /// If in-commit timestamps is not supported, or not enabled, this returns `None`.
-    /// If in-commit timestams is enabled, but the enablement version or timestamp is not present,
-    /// this returns an error.
+    /// Returns an error if only one of the enablement properties is present, as this indicates
+    /// an inconsistent state.
     #[allow(unused)]
-    pub(crate) fn in_commit_timestamp_enablement(&self) -> DeltaResult<Option<(Version, i64)>> {
+    pub(crate) fn in_commit_timestamp_enablement(
+        &self,
+    ) -> DeltaResult<InCommitTimestampEnablement> {
         if !self.is_in_commit_timestamps_enabled() {
-            return Ok(None);
+            return Ok(InCommitTimestampEnablement::NotEnabled);
         }
+
         let enablement_version = self
             .table_properties()
             .in_commit_timestamp_enablement_version;
         let enablement_timestamp = self
             .table_properties()
             .in_commit_timestamp_enablement_timestamp;
 
-        let ict_error = |err: &str| {
-            Error::generic(format!(
-                "In-commit timestamp enabled, but missing Enablement version or timestamp. {err}"
-            ))
-        };
         match (enablement_version, enablement_timestamp) {
-            (Some(version), Some(timestamp)) => Ok(Some((version, timestamp))),
-            (Some(_), None) => Err(ict_error("Enablement timestamp is not present")),
-            (None, Some(_)) => Err(ict_error("Enablement version is not present")),
-            (None, None) => Err(ict_error(
-                "Enablement version and timestamp are not present.",
+            (Some(version), Some(timestamp)) => Ok(InCommitTimestampEnablement::Enabled {
+                enablement: Some((version, timestamp))
+            }),
+            (Some(_), None) => Err(Error::generic(
+                "In-commit timestamp enabled, but enablement timestamp is missing while enablement version is present"
+            )),
+            (None, Some(_)) => Err(Error::generic(
+                "In-commit timestamp enabled, but enablement version is missing while enablement timestamp is present"
             )),
+            (None, None) => Ok(InCommitTimestampEnablement::Enabled {
+                enablement: None
+            }),
```

Comment:
Add a comment mentioning when this case happens:


```suggestion
            // If InCommitTimestamps was enabled at the beginning of the table's history,
            // it may may have an empty enablement version and timestamp
            (None, None) => Ok(InCommitTimestampEnablement::Enabled {
                enablement: None
            }),
```